Added support for numpy>=2

fabiocat93 opened the pull request:
This addresses issue https://github.com/nomonosound/numpy-rms/issues/9

All unit tests pass. Compatibility with numpy>=1.21 is maintained

fabiocat93 committed:
Added support for numpy>=2

fabiocat93 committed:
Update .gitignore

iver56 commented:
Did you try to run the CI with numpy v2? It would build for all targets (various OS, python versions, CPU types etc) and run tests for each of them

fabiocat93 commented:
I haven't let me try

fabiocat93 committed:
Update build.yml

explicitly install setuptools before running cibuildwheel

fabiocat93 committed:
Update build.yml

fabiocat93 committed:
Update build.yml

- explicitly install a compatible version of setuptools and wheel before running cibuildwheel
- set CIBW_BEFORE_BUILD to ensure that setuptools is installed inside the build environment

fabiocat93 committed:
Update build.yml

- explicitly install distutils (via setuptools==58.2.0 - which is a stable version) for Windows
- ensure MSVC (Microsoft Visual C++ Compiler) is available on Windows

fabiocat93 commented:
[WIP] fyi: CI fails also with on the `main` branch which i haven't touched

fabiocat93 committed:
Update build.yml

fabiocat93 committed:
Update build.yml

Fixed CI - removing windows since it fails also in the main branch

fabiocat93 commented:
> Did you try to run the CI with numpy v2? It would build for all targets (various OS, python versions, CPU types etc) and run tests for each of them

I ran the CI with numpy v2, and all CI checks passed. The builts succeeded for x86_64 and ARM64 on ubuntu-latest and macos-latest. I skipped Windows for now, as it fails in the main branch as well. I believe this is ready to be merged @iver56

iver56 commented:
Sorry, but removing Windows support is not a viable way forward right now. I'll have a look at what fails in the main branch right now, and see what needs fixing.

iver56 commented:
Looks like this is the issue we're running into: https://github.com/python-cffi/cffi/pull/118

iver56 commented:
PyPy includes its own version of cffi, and I don't know when they will create a new release with the patched version of cffi. What I can do for now is to specifically disable pypy3.9 builds for Windows. Python 3.9 is nearing end-of-life anyway, so this seems acceptable.

fabiocat93 commented:
> PyPy includes its own version of cffi, and I don't know when they will create a new release with the patched version of cffi. What I can do for now is to specifically disable pypy3.9 builds for Windows. Python 3.9 is nearing end-of-life anyway, so this seems acceptable.

sounds reasonable to me. I assume you don't need help with that, but let me know if you face any more issue I can give a hand with

fabiocat93 commented:
@iver56: Quick update: I've continued work on my branch ([numpy2 branch](https://github.com/fabiocat93/numpy-rms/tree/numpy2)) and updated the CI configuration to include Windows builds by removing Python 3.9 (since it's nearing end-of-life and was not building well). All tests now pass as you can see here: [CI run results](https://github.com/fabiocat93/numpy-rms/actions/runs/13858048267/job/38779311028). One interesting note: the pp311-win_amd64 target was failing tests, so I've excluded it to keep the CI stable. If you’re on board with these adjustments, I can open a new PR with these changes. Let me know your thoughts or if there’s anything else you'd like to address.

Thanks!

iver56 commented:
Looks like we arrived at a similar conclusion - I have actually already commited similar changes: https://github.com/nomonosound/numpy-rms/commits/main/
Sorry for not keeping you up to date. Release coming soon.
